main.tf

```diff
@@ -18,35 +18,38 @@
   Locals configuration for module logic
  *****************************************/
 locals {
-  organization                 = "${var.organization_id != ""}"
-  folder                       = "${var.folder_id != ""}"
-  project                      = "${var.project_id != ""}"
-  boolean_policy               = "${var.policy_type == "boolean" }"
-  list_policy                  = "${var.policy_type == "list" && !local.invalid_config}"
-  enforce                      = "${(var.allow_list_length > 0 || var.deny_list_length > 0) ? "" : var.enforce}"
-  exclude_folders_list_length  = "${length(compact(var.exclude_folders))}"
-  exclude_projects_list_length = "${length(compact(var.exclude_projects))}"
-  invalid_config_case_1        = "${var.deny_list_length > 0 && var.allow_list_length > 0}"
-  invalid_config_case_2        = "${(var.allow_list_length + var.deny_list_length) > 0 && (var.enforce == "true" || local.enforce == "false")}"
-  invalid_config               = "${var.policy_type == "list" && (local.invalid_config_case_1 || local.invalid_config_case_2)}"
+  organization                 = var.organization_id != null
+  folder                       = var.folder_id != null
+  project                      = var.project_id != null
+  boolean_policy               = var.policy_type == "boolean"
+  list_policy                  = var.policy_type == "list" && ! local.invalid_config
+  enforce                      = var.allow_list_length > 0 || var.deny_list_length > 0 ? null : var.enforce
+  exclude_folders_list_length  = length(compact(var.exclude_folders))
+  exclude_projects_list_length = length(compact(var.exclude_projects))
+  invalid_config_case_1        = var.deny_list_length > 0 && var.allow_list_length > 0
+  invalid_config_case_2        = var.allow_list_length + var.deny_list_length > 0 && (var.enforce == true || local.enforce == false)
```

Comment:
Please change (var.enforce == true || local.enforce == false)
to
var.enforce != null

Reply:
Thank you, Yurii! Makes sense! Those kind of things are rudiments of the original logic where enforce was intended to be one of `true`, `false` or `""`. We can get rid of the `null` and `local.enforce` with proper refactoring(not just syntax & types for tf 12)
fixed in [f693fa9](https://github.com/terraform-google-modules/terraform-google-org-policy/pull/11/commits/f693fa90cb7543573da98b830f2d84122823d62c)